frontend/src/app/services/utils.ts

```diff
@@ -27,11 +22,8 @@ function clippedRandn() {
 }
 
 function randn() {
-  let u = 0;
-  let v = 0;
-
-  while (u === 0) u = Math.random(); // Converting [0,1) to (0,1)
-  while (v === 0) v = Math.random();
+  const u = 1 - Math.random();
```

Comment:
@daeglin I believe this is ok, yet, pls, check: `u,v ∈ (0, 1]` (so 1 is allowed from this change on)

Reply:
Yes, that's ok